rosbag2_transport/src/rosbag2_transport/topic_filter.hpp

```diff
@@ -12,31 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef ROSBAG2_TRANSPORT__GENERIC_PUBLISHER_HPP_
-#define ROSBAG2_TRANSPORT__GENERIC_PUBLISHER_HPP_
+#ifndef ROSBAG2_TRANSPORT__TOPIC_FILTER_HPP_
+#define ROSBAG2_TRANSPORT__TOPIC_FILTER_HPP_
 
-#include <memory>
+#include <map>
 #include <string>
-
-#include "rclcpp/rclcpp.hpp"
+#include <unordered_map>
+#include <vector>
 
 namespace rosbag2_transport
 {
 
-class GenericPublisher : public rclcpp::PublisherBase
+class TopicFilter
```

Comment:
nitpick: could this just be a namespace? do you have thoughts about it being stateful in the future?